spring-boot-admin-server/src/test/java/de/codecentric/boot/admin/server/web/reactive/DataBufferTest.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2014-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package de.codecentric.boot.admin.server.web.reactive;
+
+import java.nio.ByteBuffer;
+
+import io.netty.buffer.PooledByteBufAllocator;
+import org.junit.jupiter.api.Test;
+import org.springframework.core.io.buffer.DataBuffer;
+import org.springframework.core.io.buffer.DataBufferUtils;
+import org.springframework.core.io.buffer.DefaultDataBufferFactory;
+import org.springframework.core.io.buffer.NettyDataBufferFactory;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Test to illustrate the usage of how to cache org.springframework.http.ReactiveHttpInputMessage#getBody()
+ *
+ * @see InstancesProxyController#endpointProxy(java.lang.String, org.springframework.http.server.reactive.ServerHttpRequest)
+ */
+public class DataBufferTest {
```

Comment:
What ist the point of this test? It doesn't test any of our classes...

Reply:
This shows why we should change
 `
			DataBuffer wrap = this.bufferFactory.wrap(b.asByteBuffer());
			DataBufferUtils.release(b);
			return wrap;
`
to
`			int readableByteCount = b.readableByteCount();
			DataBuffer dataBuffer = this.bufferFactory.allocateBuffer(readableByteCount);
			dataBuffer.write(b.asByteBuffer());
			DataBufferUtils.release(b);
			return dataBuffer;
`
we need to release the buffer to avoid memory leak. but `bufferFactory.wrap(b.asByteBuffer());` share the underground bytes, but when release cause share the free bytes.